File: apigateway/models/runtest.go
package models

import "time"

// RunTest tests
type RunTest struct {
	ID        uint       `json:"id" gorm:"primaryKey"`
	TestID    uint       `json:"test_id" gorm:"test_id"`
	StartTime *time.Time `json:"start_time" gorm:"start_time"`
	EndTime   *time.Time `json:"end_time" gorm:"end_time"`
	Passed    bool       `json:"passed" gorm:"passed"`
}

File: web/src/api/entity/runtest.ts
import { Query } from "../../components/basic/query";
import { makeReq, QueryString } from "../api";

export interface RunTest {
  id?: number;
  test_id: number;
  start_time: Date;
  end_time: Date;
  passed: boolean;
}

export const GetRunTest = async (runTest: RunTest) => {
  return await makeReq(`/run_test/${runTest.id}`, "GET");
};

export const listRunTest = async () => {
  return await makeReq("/run_test", "Get");
};

export const deleteRunTest = async (runTest: RunTest) => {
  return await makeReq("/run_test", "DELETE", runTest);
};

export const listRunTestsOfTest = (testID: number) => async (query?: Query) => {
  return await makeReq(`/test/${testID}/run_tests?${QueryString(query)}`);
};
